plugins/httpapi/aci.py

```diff
@@ -0,0 +1,384 @@
+        data = json.dumps(payload)
+        try:
+            response, response_data = self.connection.send(path, data, method=method)
+        except Exception as exc_logout:
+            msg = "Error on attempt to logout from APIC. {0}".format(exc_logout)
+            raise ConnectionError(self._return_info(None, method, path, msg))
+        self.connection._auth = None
+        self._verify_response(response, method, path, response_data)
+
+    def set_parameters(self, method, path, data):
+        if self.params.get("port") is not None:
+            self.connection.set_option("port", self.params.get("port"))
+
+        if self.params.get("username") is not None:
+            self.connection.set_option("remote_user", self.params.get("username"))
+
+        if self.params.get("use_proxy") is not None:
+            self.connection.set_option("use_proxy", self.params.get("use_proxy"))
+
+        if self.params.get("use_ssl") is not None:
+            self.connection.set_option("use_ssl", self.params.get("use_ssl"))
+
+        if self.params.get("validate_certs") is not None:
+            self.connection.set_option("validate_certs", self.params.get("validate_certs"))
+
+        # The command timeout which is the response timeout from APIC
+        # If the persistent_connect_timeout is less than the response timeout from APIC the persistent socket connection will fail
+        if self.params.get("timeout") is not None:
+            self.connection.set_option("persistent_command_timeout", self.params.get("timeout"))
+            self.connection.set_option("persistent_connect_timeout", self.params.get("timeout") + 30)
+
+        # Start with certificate authentication
+        if self.auth is not None:
+            # If session_key is present in the inventory, certificate in the task is ignored. In order to avoid this, we explicitly set session_key to None
+            # to give preference to credentials/certificate at the task level.
+            if self.connection.get_option("session_key") is not None:
+                self.connection.set_option("session_key", None)
+            # Check if the previous task was running with the credentials at the task level/inventory or the certificate in the inventory
+            if self.check_auth_from_credentials_task or self.check_auth_from_credentials_inventory or self.check_auth_from_private_key_inventory:
+                self.host_counter = 0
+                self.check_auth_from_credentials_task = False
+                self.check_auth_from_credentials_inventory = False
+                self.check_auth_from_private_key_inventory = False
+            self.check_auth_from_private_key_task = True
+            self.connection._auth = {"Cookie": "{0}".format(self.auth)}
+            self.connection.queue_message("step:", "Setting certificate authentication at task level")
+            # Override parameter in @ensure_connect
+            self.connection._connected = True
```

Comment:
Is this set to True in order to always skip login because it is certificate based auth?